src/main/java/com/shaft/tools/tms/XrayIntegrationManager.java

```diff
@@ -0,0 +1,111 @@
+     public static String _JiraAuthorization=
+             Base64.getEncoder().encodeToString(System.getProperty("authorization").getBytes());
+     public static String _ProjectKey= System.getProperty("projectKey");
+     public static String _TestExecutionID=null;
+
+     public static void setup() {
+         baseURI = System.getProperty("jiraUrl");
+         given()
+                 .config(RestAssuredConfig.config().sslConfig(SSLConfig.sslConfig().allowAllHostnames()))
+                 .relaxedHTTPSValidation();
+     }
+
+    public static void importCucumberResults(String filepath) throws Exception {
+
+        setup();
+        String reportPath = System.getProperty("user.dir")+"\\"+filepath;
+        ReportManager.logDiscrete("uploading file: "+reportPath);
+        ReportManager.logDiscrete("Length: "+new File(reportPath).length());
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        JsonElement je = JsonParser.parseString( new String(Files.readAllBytes(Paths.get(reportPath))));
+        String prettyJsonString = gson.toJson(je);
+        ReportManager.logDiscrete("Pretty: \n"+ prettyJsonString);
+
+        try {
+            Response response =given()
+                    .contentType("application/json")
+                    .header("Authorization", "Basic " + _JiraAuthorization)
+                    .body(prettyJsonString)
+                    .expect().statusCode(200)
+                    .when()
+                    .post( "/rest/raven/1.0/import/execution/cucumber").then().extract().response();
+            ReportManager.logDiscrete("#################"+ response.asString());
+            _TestExecutionID=response.jsonPath().get("testExecIssue.key").toString();
+            ReportManager.logDiscrete("ExecutionID: "+_TestExecutionID);
+        }catch (Exception e){
+            ReportManagerHelper.log(e);
+        }
+    }
+
+    public static void renameTestExecutionSuit(String executionName, String executionDescription) {
+         if (_TestExecutionID==null) return;
+         setup();
+        Calendar cal = Calendar.getInstance();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+         String body= "{\r\n    \"fields\" : {\r\n       " +
+                 " \"summary\": " +
+                 "\"Execution results "+executionName+" | "+sdf.format(cal.getTime())+"\",\r\n        " +
+                 "\"description\": " +
+                 "\""+executionDescription+"\"\r\n    }\r\n}";
+        try {
+            Response response =given()
+                    .contentType("application/json")
+                    .header("Authorization", "Basic " + _JiraAuthorization)
+                    .body(body)
+                    .expect().statusCode(204)
+                    .when()
+                    .put( "/rest/api/2/issue/" +_TestExecutionID).then().extract().response();
+            ReportManager.logDiscrete("#################"+ response.asString());
+
```

Comment:
There's no need to report the response body in the discrete log... It will flood it significantly.
As an enhancement we can use SHAFT's built-in requests as they're more ready for logging and reporting.

Reply:
Done.